isolates/bin/long_running_isolate.dart

```diff
@@ -0,0 +1,88 @@
+// Spawn an isolate, read multiple files, send their contents to the spawned
+// isolate, and wait for the parsed JSON.
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:isolate';
+
+final fileNames = [
+  'json_01.json',
+  'json_02.json',
+  'json_03.json',
+];
+
+// Reads multiple files and parses them on a separate isolate.
+Future<void> main() async {
+  // Read some data.
+  final fileData = await _readFiles();
+
+  // Use that data.
+  await for (final jsonData in _parseJsonStrings(fileData)) {
+    print('Received JSON with ${jsonData.length} keys');
+  }
+}
+
+// Reads multiple files simultaneously using Future.wait().
+Future<List<String>> _readFiles() async {
+  return await Future.wait([
+    ...fileNames.map((f) => File(f)).map((file) => file.readAsString()),
```

Comment:
```suggestion
    ...filenames.map((f) => File(f)).map((file) => file.readAsString()),
```

Reply:
`filename` is more commonly used I think